File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/incoming/AuthenticatorAaidMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.incoming

import kotlinx.serialization.Serializable

@Serializable
data class AuthenticatorAaidMessage(
    override val operationId: String,
    val aaid: String
) : ChannelInMessage()

File: ios/Model/Message/Incoming/AuthenticatorAaidMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import Foundation

class AuthenticatorAaidMessage: ChannelInMessage {
	// MARK: Properties

	let aaid: String

	// MARK: Decodable

	enum CodingKeys: String, CodingKey {
		case aaid
	}

	required init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		self.aaid = try container.decode(String.self, forKey: .aaid)
		try super.init(from: decoder)
	}
}
